### PR TITLE
Adding GetAttachState and SetAttachState to disk api group

### PR DESCRIPTION
**Pull request by jmpfar**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
1. Adding v1beta2 disk api group
2. Adding SetAttachState, a function that allows changing the disk's online/offline status. 
    This is needed because SanPolicy is not set to automount by default on all OSes. 
    Consequently iSCSI disks will not be set online on connection.
3. Adding GetAttachState, a function that queries a disk online/offline status
4. Adding an integration test 

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Add GetAttachState/SetAttachState API that handle online/offline state of disk
```

### CODE REVIEW

File: integrationtests/disk_test.go [hunk 1]
```diff
@@ -6,22 +6,22 @@ import (
...
-	"github.com/kubernetes-csi/csi-proxy/client/api/disk/v1beta1"
-	v1beta1client "github.com/kubernetes-csi/csi-proxy/client/groups/disk/v1beta1"
+	"github.com/kubernetes-csi/csi-proxy/client/api/disk/v1beta2"
+	v1beta2client "github.com/kubernetes-csi/csi-proxy/client/groups/disk/v1beta2"
 	"github.com/stretchr/testify/require"
 )
 
 // This test is meant to run on GCE where the page83 ID of the first disk contains
 // the host name
 func TestDiskAPIGroupV1Beta1(t *testing.T) {
```
Comment (ddebroy) [inline integrationtests/disk_test.go:16]: Please rename `TestDiskAPIGroupV1Beta1` to `TestDiskAPIGroup` as the version will continue to evolve. 

Can you also please add a couple of tests for the new online/offline functionality. The test can [1] set the disk offline through powershell directly [2] try to online it through the csi-proxy API and [3] verify disk online status through powershell. The same test can be repeated to offline a disk through csi-proxy API.

File: client/api/disk/v1beta2/api.proto [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    rpc Rescan(RescanRequest) returns (RescanResponse) {}
+
+    // ListDiskIDs returns a map of DiskID objects where the key is the disk number
+    rpc ListDiskIDs(ListDiskIDsRequest) returns (ListDiskIDsResponse) {}
+
+    // DiskStats returns the stats for the disk
+    rpc DiskStats(DiskStatsRequest) returns (DiskStatsResponse) {}
+
+    // SetAttachState sets the offline/online state of a disk
+    rpc SetAttachState(SetAttachStateRequest) returns (SetAttachStateResponse) {}
```
Comment (ddebroy) [inline client/api/disk/v1beta2/api.proto:26]: Can you also add a `GetAttachState` please?

**Review by ddebroy — COMMENTED** [review]
Thanks for the PR @jmpfar. A couple of suggestions below. Otherwise overall structure looks great!

File: integrationtests/disk_test.go (hunk 1 above)
Comment (jmpfar) [reply integrationtests/disk_test.go]: Done

File: client/api/disk/v1beta2/api.proto (hunk 2 above)
Comment (jmpfar) [reply client/api/disk/v1beta2/api.proto:26]: Added, PTAL

**Review by ddebroy — COMMENTED** [review]
Thanks for the changes and tests!

/lgtm
/approve

**Pull request merged**